src/table/components/__tests__/TableTotals.spec.jsx

```diff
@@ -0,0 +1,152 @@
+      });
+
+      it('should return bottom when the position is set to bottom', () => {
+        layout.totals.position = 'bottom';
+        expect(getTotalPosition(layout)).toBe('bottom');
+      });
+
+      it('should not show totals when the position is set to be visible but there is no grandTotal', () => {
+        layout.qHyperCube.qGrandTotalRow.length = 0;
+        expect(getTotalPosition(layout)).toBe('noTotals');
+        const { queryByText } = renderTableTotals();
+        expect(queryByText(tableData.columns[0].totalInfo)).not.toBeInTheDocument();
+        expect(queryByText(tableData.columns[1].totalInfo)).not.toBeInTheDocument();
+      });
+
+      it('should not show totals when position is set to be visible but table has only dimension', () => {
+        tableData = {
+          columns: [{ id: 0, isDim: true, dataColIdx: 0, totalInfo: 'Totals' }],
+        };
+        layout.qHyperCube.qMeasureInfo.length = 0;
+        expect(getTotalPosition(layout)).toBe('noTotals');
+        const { queryByText } = renderTableTotals();
+        expect(queryByText(tableData.columns[0].totalInfo)).not.toBeInTheDocument();
+      });
+
+      it('should show totals when table has only measure', () => {
+        tableData = {
+          columns: [{ id: 0, isDim: false, dataColIdx: 0, totalInfo: '350' }],
+        };
+        layout.qHyperCube.qDimensionInfo.length = 0;
+        layout.totals.position = 'top';
+        expect(getTotalPosition(layout)).toBe('top');
+        const { queryByText } = renderTableTotals();
+        expect(queryByText(tableData.columns[0].totalInfo)).toHaveTextContent('350');
+      });
+    });
+
+    describe('When total auto mode is on', () => {
+      it('should show totals at top by default', () => {
+        layout.totals.show = true;
+        layout.totals.position = 'noTotals';
+        expect(getTotalPosition(layout)).toBe('top');
+        const { queryByText } = renderTableTotals();
+        expect(queryByText(tableData.columns[0].totalInfo)).toHaveTextContent('Totals');
+        expect(queryByText(tableData.columns[1].totalInfo)).toHaveTextContent('350');
+      });
+
+      it('should return not show totals when table has only measure', () => {
+        tableData = {
+          columns: [{ id: 0, isDim: false, dataColIdx: 0, totalInfo: '350' }],
+        };
+        layout.totals.show = true;
+        layout.qHyperCube.qDimensionInfo.length = 0;
+        expect(getTotalPosition(layout)).toBe('noTotals');
+        const { queryByText } = renderTableTotals();
+        expect(queryByText(tableData.columns[0].totalInfo)).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Handle keys call when TableTotals is rendered', () => {
+    beforeEach(() => {
+      layout.totals.position = 'top';
+    });
+    it('should call totalHandleKeyPress when keyDown on a total cell', () => {
+      jest.spyOn(handleKeyPress, 'totalHandleKeyPress').mockImplementation(() => jest.fn());
```

Comment:
make sure you clear this in an afterEach: `afterEach(() => jest.clearAllMocks());`